### PR TITLE
extend secure store with test profile

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
We need to sure the new Test Profil in the App. Added a new model to store the profile, also added some unit tests

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6629

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Extensions/ISO8601DateFormatter+extentions.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+extension ISO8601DateFormatter {
+
+	static let justDate: ISO8601DateFormatter = {
```
Comment (SwissMountainFrog) [inline src/xcode/ENA/ENA/Source/Extensions/ISO8601DateFormatter+extentions.swift:9]: Maybe some doc comments about what `justDate` is/should be used for

File: src/xcode/ENA/ENA/Source/Models/Profile/RapidTestProfile.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+//
+
+import Foundation
+
+struct RapidTestProfile: Codable {
+
+	// MARK: - Init
+
+	init(
+		forename: String? = nil,
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Models/Profile/RapidTestProfile.swift:12]:
```suggestion
		firstName: String? = nil,
```
More commonly used and also consistent with TestedPerson on AntigenTest

**Review by marcussc — APPROVED** [review]

**Review by SwissMountainFrog — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Workers/Store/Store.swift [hunk 3]
```diff
@@ -181,6 +181,8 @@ protocol CoronaTestStoringLegacy {
 
 	var isSubmissionConsentGiven: Bool { get set }
 
+	var profile: RapidTestProfile? { get set }
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Workers/Store/Store.swift:184]:
```suggestion
	var anitgenTestProfile: RapidTestProfile? { get set }
```

Maybe to make it more clear as there is so much stuff in the store.

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by nickguendling — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Workers/Store/SecureStore.swift [hunk 4]
```diff
@@ -426,6 +426,11 @@ extension SecureStore: CoronaTestStoringLegacy {
 		set { kvStore["isSubmissionConsentGiven"] = newValue }
 	}
 
+	var anitgenTestProfile: AntigenTestProfile? {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Workers/Store/SecureStore.swift:429]: antigenTestProfile

**Review by naveeddotio — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Workers/Store/Store.swift [hunk 5]
```diff
@@ -177,6 +177,8 @@ protocol CoronaTestStoringLegacy {
 
 	var isSubmissionConsentGiven: Bool { get set }
 
+	var anitgenTestProfile: AntigenTestProfile? { get set }
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Workers/Store/Store.swift:180]: typo here as well

File: src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift [hunk 6]
```diff
@@ -115,7 +115,7 @@ final class MockTestStore: Store, PPAnalyticsData {
 	var lastSuccessfulSubmitDiagnosisKeyTimestamp: Int64?
 	var positiveTestResultWasShown: Bool = false
 	var isSubmissionConsentGiven = false
-
+	var anitgenTestProfile: AntigenTestProfile?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift:118]: antigen

File: src/xcode/ENA/ENA/Source/Workers/Store/SecureStore.swift (hunk 4 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Workers/Store/SecureStore.swift]: done

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**